src/entities/CheckScenes/task/taskRunnerSqs.ts

```diff
@@ -227,3 +233,50 @@ export async function taskRunnerSqs(job: DeploymentToSqs) {
     CheckScenesModel.createMany(placesToDisable)
   }
 }
+
+async function getValidCategories(creatorTags: string[]) {
+  const forbidden = ["poi", "featured"]
+
+  const availableCategories = await CategoryModel.findActiveCategories()
+
+  const validCategories = []
+
+  for (const tag of creatorTags) {
+    if (forbidden.includes(tag)) continue
+
+    if (availableCategories.find(({ name }) => name === tag)) {
+      validCategories.push(tag)
+    }
+
+    if (validCategories.length === 3) break
+  }
+
+  return validCategories
+}
+
+async function overridePlaceCategories(placeId: string, creatorTags: string[]) {
+  if (!creatorTags.length) return
+
+  const validCategories = await getValidCategories(creatorTags)
+
+  if (!validCategories.length) return
+
+  const currentCategories = await PlaceCategories.findCategoriesByPlaceId(
+    placeId
+  )
+
+  if (currentCategories.find(({ category_id }) => category_id === "poi")) {
+    validCategories.push("poi")
+  }
+
+  if (currentCategories.find(({ category_id }) => category_id === "featured")) {
+    validCategories.push("featured")
+  }
```

Comment:
What do you think about `new Set(...)` and the  `currentCategories .has(SomeConstant.Poi)`